minerva-core/src/test/java/org/geneontology/minerva/BlazegraphMolecularModelManagerTest.java

```diff
@@ -0,0 +1,110 @@
+package org.geneontology.minerva;
+
+import static org.junit.Assert.*;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.rdf.model.StmtIterator;
+import org.junit.Test;
+import org.semanticweb.owlapi.model.OWLOntologyCreationException;
+
+import owltools.OWLToolsTestBasics;
+import owltools.graph.OWLGraphWrapper;
+import owltools.io.CatalogXmlIRIMapper;
+import owltools.io.ParserWrapper;
+
+public class BlazegraphMolecularModelManagerTest extends OWLToolsTestBasics {
+
+	/**
+	 * Test whether the revised import function properly digest turtle files. We import a ttl file 
+	 * into the model, dump the model into file, and then compare the initial file with dumped ones. 
+	 * Check this pull request for more information: https://github.com/geneontology/minerva/pull/144/files 
+     *
+	 * @throws OWLOntologyCreationException
+	 * @throws IOException
+	 */
+	@SuppressWarnings("rawtypes")
+	@Test
+	public void testImportDump() throws OWLOntologyCreationException, IOException {
+		/* A temporary journal file for Blazegraph storage system */
+		String journalPath = "/tmp/journal.jrnl";	
+		/* I used the file from one of the turtle file in
+		 * https://github.com/geneontology/noctua-models/blob/master/models/0000000300000001.ttl */
+		String sourceModelPath = "src/test/resources/dummy-noctua-model.ttl";
+		/* It seems that the path generated by dumpAllStoredModels is not properly escaped. */
+		String targetModelPath = "/tmp/http:/model.geneontology.org/0000000300000001.ttl";
+
+		/* Delete any temporary files if they already exist */
+		File journalFile = new File(journalPath);
+		File targetModelFile = new File(journalPath);
+		FileUtils.deleteQuietly(journalFile);
+		FileUtils.deleteQuietly(targetModelFile);
+
+		final ParserWrapper pw1 = new ParserWrapper();
+		pw1.addIRIMapper(new CatalogXmlIRIMapper(new File("src/test/resources/mmg/catalog-v001.xml")));
+		OWLGraphWrapper g = pw1.parseToOWLGraph(getResourceIRIString("mmg/basic-tbox-importer.omn"));
+		BlazegraphMolecularModelManager m3 = new BlazegraphMolecularModelManager<>(g, "go", journalPath, "/tmp");
```

Comment:
JUnit has some methods for more safely making tmp folders. There is an example in `UndoAwareMolecularModelManagerTest`. I'm not sure how much this matters.

Reply:
ok sure I will check that test and see if I can use method you suggested.